plaidml2/bridge/keras/__init__.py

```diff
@@ -295,7 +356,13 @@ def bias_add(x, bias, data_format=None):
 
 
 def binary_crossentropy(target, output, from_logits=False):
-    _report_unimplemented('binary_crossentropy')
+    logger.debug('binary_crossentropy(target: {}, output: {}, from_logits: {})'.format(
```

Comment:
I'll take a quick look and see if I see something straightforward. I'm not sure whether there'll be a nice programmatic way to get at the parameter names, which seems core to having a decorator be simpler & more consistent

Reply:
Use the `inspect` module.  `inspect.signature(some_callable).parameters` has the info.

Reply:
Decorator++
Adding another dependency--

Reply:
It looks like inspect is a core part of python (unless I misunderstand how that works?): https://docs.python.org/3/library/inspect.html

Reply:
My comment is about adding new pypi dependencies. If we can make it work with core python, that's good. Does the `inspect` module add much overhead? It'd be best if the overhead would only be paid if logging at debug mode is enabled.

Reply:
I wrapped most of the decorator code in a `if logger.isEnabledFor(logging.DEBUG):` block. The only extra overhead when debug is off is in calling a wrapping decorator that calls the function rather than calling the function directly.